File: content/LLD/Reference-Code/5. Strategy Pattern/main.py
from discount_service import DiscountService
from diawli import DiwaliStrategy
from holi import HoliStrategy


diwali_strategy = DiwaliStrategy()
holi_strategy = HoliStrategy()

discount_service = DiscountService(diwali_strategy)
discount_service.process()

discount_service.set_strategy(holi_strategy)
discount_service.process()

File: content/LLD/Reference-Code/5. Strategy Pattern/Main.java
public class Main {
    public static void main(String[] args) {
        DiscountStrategy diwaliStrategy = new DiwaliStrategy();
        DiscountStrategy holiStrategy = new HoliStrategy();

        DiscountService discountService = new DiscountService(diwaliStrategy);
        discountService.process();

        discountService.setStrategy(holiStrategy);
        discountService.process();
    }
}
